types/list.go

```diff
@@ -54,22 +47,50 @@ type ListBucketsResult struct {
 // ObjectMeta is the structure for metadata service user object
 type ObjectMeta struct {
 	// object_info defines the information of the object.
-	ObjectInfo *GetObjectResult `json:"object_info"`
+	ObjectInfo *ObjectInfo `json:"object_info"`
 	// locked_balance defines locked balance of object
 	LockedBalance string `json:"locked_balance"`
 	// removed defines the object is deleted or not
 	Removed bool `json:"removed"`
+	// update_at defines the block number when the object updated
+	UpdateAt int64 `json:"update_at,string"`
+	// delete_at defines the block number when the object deleted
+	DeleteAt int64 `json:"delete_at,string"`
+	// delete_reason defines the deleted reason of object
+	DeleteReason string `json:"delete_reason"`
+	// operator defines the operator address of object
+	Operator string `json:"operator"`
+	// create_tx_hash defines the creation transaction hash of object
+	CreateTxHash string `json:"create_tx_hash"`
+	// update_tx_hash defines the update transaction hash of object
+	UpdateTxHash string `json:"update_tx_hash"`
+	// seal_tx_hash defines the sealed transaction hash of object
+	SealTxHash string `json:"seal_tx_hash"`
 }
 
 // BucketMeta is the structure for metadata service user bucket
 type BucketMeta struct {
 	// bucket_info defines the information of the bucket.
-	BucketInfo *BucketInfo `protobuf:"bytes,1,opt,name=bucket_info,json=bucketInfo,proto3" json:"bucket_info"`
+	BucketInfo *BucketInfo `json:"bucket_info"`
 	// removed defines the bucket is deleted or not
-	Removed bool `protobuf:"varint,2,opt,name=removed,proto3" json:"removed"`
+	Removed bool `json:"removed"`
+	// delete_at defines the block number when the bucket deleted.
+	DeleteAt int64 `json:"delete_at,string"`
+	// delete_reason defines the deleted reason of bucket
+	DeleteReason string `json:"delete_reason"`
+	// operator defines the operator address of bucket
+	Operator string `json:"operator"`
+	// create_tx_hash defines the creation transaction hash of object
+	CreateTxHash string `json:"create_tx_hash"`
+	// update_tx_hash defines the update transaction hash of object
+	UpdateTxHash string `json:"update_tx_hash"`
+	// update_at defines the block number when the object updated
+	UpdateAt int64 `json:"update_at,string"`
+	// update_time defines the block number when the object updated
+	UpdateTime int64 `json:"update_time,string"`
 }
 
-// ObjectInfo differ from GetObjectResult in greenfield as it adds uint64/int64 unmarshal guide in json part
+// ObjectInfo differ from ObjectInfo in greenfield as it adds uint64/int64 unmarshal guide in json part
 type ObjectInfo struct {
```

Comment:
Why not use the `ObjectInfo` in greenfield repo?

https://github.com/bnb-chain/greenfield/blob/1fbd424604e0b25572669f4ce1d944e7449f9c99/x/storage/types/types.pb.go#L278-L305

Reply:
As explained in the comment, uint64/int64 unmarshal cannot be supported correctly with greenfield repo's objectInfo.

Reply:
We now remove the amino codec and use proto codec. Could you re-check this? 

The below is an test result of amino and proto codec.
```
=== RUN   TestAminoAndProto
    message_test.go:705: Amino codec: {"bucket_name":"testbucket","content_type":"content-type","creator":"0xc1d0b908adb7323521C86B36576b26C24976713C","expect_checksums":["06FX+nebsnJ2vQr/CvdxlC0l1g6J+qIZqD50Pq/nTps=","XdG6EqrzdegOrnDaLFsPNg2Rfflw+wZDCJh9YX3BMyA=","MVM4xtu96vChuBh9M8IPlg5c2pdgDQnb20+lUYdBtbc=","5tE3GWIciPWpYnEK+N1Ip01RHaIhJbDzULrLNindKf0=","cv/2J/Cv0+FRr+YjkALJA69Qcuir1s+VgLF4VUhf3lY=","5H8i410dBWvjzNLsKR7RlsD6VWlNbMhIT9YkTyfTuvU=","nIjcygk8N0W2vukSJXGQjMlveivDLG5NAckpgqR7CIw="],"expect_secondary_sp_addresses":["0x64553d63fcB9A6606F7647276D3b6c397D61578A","0x98A05FD977241b35fF65f6a6a4dc1e29294831B9","0x0B62BE4BA8cC9423dBfB6093223F6579D5B7b526","0xC39987AA9b7bAC094A1F0328608AB3ceD1329344","0xAE5FDE6D278b8934937C0cf89fb7fD065dF31746","0xD835f9445fccFE9A9c9E9E21691A641e308C31d0"],"object_name":"testobject","payload_size":"1024","primary_sp_approval":{},"visibility":2}
    message_test.go:706: Proto codec: {"bucket_name":"testbucket","content_type":"content-type","creator":"0xc1d0b908adb7323521C86B36576b26C24976713C","expect_checksums":["06FX+nebsnJ2vQr/CvdxlC0l1g6J+qIZqD50Pq/nTps=","XdG6EqrzdegOrnDaLFsPNg2Rfflw+wZDCJh9YX3BMyA=","MVM4xtu96vChuBh9M8IPlg5c2pdgDQnb20+lUYdBtbc=","5tE3GWIciPWpYnEK+N1Ip01RHaIhJbDzULrLNindKf0=","cv/2J/Cv0+FRr+YjkALJA69Qcuir1s+VgLF4VUhf3lY=","5H8i410dBWvjzNLsKR7RlsD6VWlNbMhIT9YkTyfTuvU=","nIjcygk8N0W2vukSJXGQjMlveivDLG5NAckpgqR7CIw="],"expect_secondary_sp_addresses":["0x64553d63fcB9A6606F7647276D3b6c397D61578A","0x98A05FD977241b35fF65f6a6a4dc1e29294831B9","0x0B62BE4BA8cC9423dBfB6093223F6579D5B7b526","0xC39987AA9b7bAC094A1F0328608AB3ceD1329344","0xAE5FDE6D278b8934937C0cf89fb7fD065dF31746","0xD835f9445fccFE9A9c9E9E21691A641e308C31d0"],"object_name":"testobject","payload_size":"1024","primary_sp_approval":{"expired_height":"0","sig":null},"redundancy_type":"REDUNDANCY_EC_TYPE","visibility":"VISIBILITY_TYPE_PRIVATE"}
--- PASS: TestAminoAndProto (0.00s)
PASS
```

Reply:
Thanks for letting me know. What I'm trying to say here is that amino and proto module both have interface like: MustUnmarshalJSON(bz []byte, ptr proto.Message)
Here when we unmarshal, the struct we are unmarshaling are actually objectMeta and bucketMeta, which are not themselves proto messages. If we want to use the amino or proto module, we need to either import SP repo or copy the pb file from SP repo, which are more costly than just maintaining all the struct locally.

Reply:
Update: will upload the SP metadata's proto on to cloud buf registry, and use them directly from there in next phase.

Reply:
Thanks. I get it. I think we need to review the dependencies between greenfield-go-sdk/greenfield/greenfield-storage-provider and try to decouple them as much as possible. 


BTW, It is not urgent, but necessary.